code/go/0chain.net/blobbercore/challenge/protocol.go

```diff
@@ -70,49 +73,60 @@ func (cr *ChallengeEntity) LoadValidationTickets(ctx context.Context) error {
 		cr.StatusMessage = "No validators assigned to the challenge"
 		cr.UpdatedAt = time.Now().UTC()
 
+		logging.Logger.Debug(cr.StatusMessage + "for challenge: " + cr.ChallengeID)
+
 		if err := cr.Save(ctx); err != nil {
-			Logger.Error("[challenge]db: ", zap.String("challenge_id", cr.ChallengeID), zap.Error(err))
+			logging.Logger.Error("[challenge]db: ", zap.String("challenge_id", cr.ChallengeID), zap.Error(err))
+			return err
 		}
-		return common.NewError("no_validators", "No validators assigned to the challenge")
+		return nil
 	}
 
 	allocationObj, err := allocation.GetAllocationByID(ctx, cr.AllocationID)
 	if err != nil {
 		return err
 	}
 
+	// Lock allocation changes from happening in handler.CommitWrite function
+	// This lock should be unlocked as soon as possible. We should not defer
+	// unlocking it as it will be locked for longer time and handler.CommitWrite
+	// will fail.
+	allocMu := lock.GetMutex(allocationObj.TableName(), allocationObj.ID)
+	allocMu.Lock()
+
 	wms, err := writemarker.GetWriteMarkersInRange(ctx, cr.AllocationID, cr.AllocationRoot, allocationObj.AllocationRoot)
 	if err != nil {
+		allocMu.Unlock()
 		return err
 	}
 	if len(wms) == 0 {
+		allocMu.Unlock()
 		return common.NewError("write_marker_not_found", "Could find the writemarker for the given allocation root on challenge")
 	}
 
 	rootRef, err := reference.GetReference(ctx, cr.AllocationID, "/")
+	if err != nil {
+		allocMu.Unlock()
+		cr.ErrorChallenge(ctx, err)
+		return err
+	}
+
 	blockNum := int64(0)
 	if rootRef.NumBlocks > 0 {
```

Comment:
in numBlocks == 0, shouldn't we stop process here?

Reply:
No.
This is the actual fix for blank allocation.

Even if there is not files in allocation, it should still provide validators with list of writemarkers so that validator can verify the challenge.